shuffler/src/main/java/com/shuffle/p2p/MarshallChannel.java

```
@@ -44,7 +44,7 @@ public boolean closed() {
         @Override
         public boolean send(X x) throws InterruptedException, IOException {
             Bytestring b = m.marshall(x);
-            return b != null && s.send(b);
+            return s.send(b);
```

b shouldn't be null. If it is ever null, that is a bug. Therefore if you think b ever will be null, then you should throw a NullPointerException rather than checking for null and returning false.